NOLF2/ClientShellDLL/ClientShellShared/ClientUtilities.cpp

```diff
@@ -366,7 +366,55 @@ LTRESULT SendEmptyServerMsg(uint32 nMsgID, uint32 nFlags)
 
 #ifndef _WIN32
 
-#define FormatMessage(a,b,c,d,e,f,g)
+#define FORMAT_MESSAGE_FROM_STRING 0
+static void FormatMessage(int flags, char* source, int message_id, int language_id, char* dest, int len, va_list* args)
```

Comment:
the dest length is never checked
the function should end early in case it has filled up the dest buffer
and place the '\0' at `dest[len-1]`

Reply:
Done.

Reply:
Thank you.

Looking at the parameter list maybe later we can make source const char*, I don't see the function  modify data through the source pointer.